rollup-interface/src/state_machine/mocks/zk_vm.rs

```diff
@@ -79,17 +78,30 @@ impl Zkvm for MockZkvm {
         Ok(proof.log)
     }
 
+    #[cfg(feature = "std")]
     fn verify_and_extract_output<
         Add: crate::RollupAddress,
         Da: crate::da::DaSpec,
-        Root: Serialize + DeserializeOwned,
+        Root: Serialize + serde::de::DeserializeOwned,
     >(
         serialized_proof: &[u8],
         code_commitment: &Self::CodeCommitment,
     ) -> Result<crate::zk::StateTransition<Da, Add, Root>, Self::Error> {
         let output = Self::verify(serialized_proof, code_commitment)?;
         Ok(bincode::deserialize(output)?)
     }
+
+    #[cfg(not(feature = "std"))]
+    fn verify_and_extract_output<
+        Add: crate::RollupAddress,
+        Da: crate::da::DaSpec,
+        Root: Serialize + serde::de::DeserializeOwned,
+    >(
+        _serialized_proof: &[u8],
+        _code_commitment: &Self::CodeCommitment,
+    ) -> Result<crate::zk::StateTransition<Da, Add, Root>, Self::Error> {
+        todo!("the current version of bincode doesn't support no-std; however, the next version is scheduled to")
+    }
 }
```

Comment:
A unified function body seems more compact and uses less feature gating magic:

```suggestion
    fn verify_and_extract_output<
        Add: crate::RollupAddress,
        Da: crate::da::DaSpec,
        Root: Serialize + serde::de::DeserializeOwned,
    >(
        serialized_proof: &[u8],
        code_commitment: &Self::CodeCommitment,
    ) -> Result<crate::zk::StateTransition<Da, Add, Root>, Self::Error> {
        if cfg!(feature = "std") {
            let output = Self::verify(serialized_proof, code_commitment)?;
            Ok(bincode::deserialize(output)?)
        } else {
            todo!("the current version of bincode doesn't support no-std; however, the next version is scheduled to")
        }
    }
```

Reply:
Tested; the `bincode` will be compiled on non-optimized builds for `no-default-features`. Since it's not present under `no_std`, it will cause a compiler error